spring-integration-file/src/main/java/org/springframework/integration/file/filters/AbstractFileListFilter.java

```diff
@@ -42,11 +42,17 @@
 		return accepted;
 	}
 
+	@Override
+	public boolean supportsSingleFileFiltering() {
+		return true;
+	}
+
 	/**
 	 * Subclasses must implement this method.
 	 * @param file The file.
 	 * @return true if the file passes the filter.
 	 */
+	@Override
 	public abstract boolean accept(F file);
```

Comment:
Can we remove this `abstract` now since you have that `default` one in the super interface already?